tvx/builders/generator.go

```diff
@@ -0,0 +1,166 @@
+package builders
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sync"
+
+	"github.com/filecoin-project/oni/tvx/schema"
+)
+
+// Generator is a batch generator and organizer of test vectors.
+//
+// Test vector scripts are simple programs (main function). Test vector scripts
+// can delegate to the Generator to handle the execution, reporting and capture
+// of emitted test vectors into files.
+//
+// Generator supports the following CLI flags:
+//
+//  -o <directory>
+//		directory where test vector JSON files will be saved.
+//
+//  -f <regex>
+//		regex filter to select a subset of vectors to execute; matched against
+//	 	the vector's ID.
+//
+// Scripts can bundle test vectors into "groups". The generator will execute
+// each group in parallel, and will write each vector in a file:
+// <output_dir>/<group>-<sequential>.json
+type Generator struct {
+	OutputPath string
+	Filter     *regexp.Regexp
+
+	wg sync.WaitGroup
+}
+
+type MessageVectorGenItem struct {
+	Metadata *schema.Metadata
+	Func     func(*Builder)
+}
+
+func NewGenerator() *Generator {
+	// Consume CLI parameters.
+	var (
+		outputDir = flag.String("o", "", "output directory")
+		filter    = flag.String("f", "", "filter regex for test vectors to generate")
+	)
+
+	flag.Parse()
+
+	ret := new(Generator)
+
+	// Output directory is compulsory.
+	if *outputDir == "" {
+		fmt.Printf("output directory not specified\n")
+		os.Exit(1)
```

Comment:
```suggestion
		log.Fatalf("output directory not specified\n")
```

Reply:
Implemented.

Context: we were suppressing logs in the `init` function of `builder.go`, to not muddy the output JSON, that's why I was using `fmt.Print*` directly. But @willscott suggested to send the log package to `os.Stderr`, which keeps Unix pipes working, but it muddies the console output. To make things better, I have prefix `log` output with `>>>`.